src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/PropertyVisibilityTests.cs

```diff
@@ -1,10 +1,60 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Text.Json.Serialization;
+using Xunit;
+
 namespace System.Text.Json.Serialization.Tests
 {
     public sealed partial class PropertyVisibilityTestsDynamic : PropertyVisibilityTests
     {
         public PropertyVisibilityTestsDynamic() : base(JsonSerializerWrapper.StringSerializer) { }
+
+        [Fact]
+        public void CorrectlyIgnorePropertyWithNewModifier()
```

Comment:
The test should run the same for both reflection and the source generator. Would it be possible to move the test to the base `PropertyVisibilityTests` class? Note that this would additionally require the following two changes:

1. Replace any direct calls to `JsonSerializer.Serialize`/`Deserialize` with the `Serializer.SerializeWrapper` test abstraction.
2. For the case of the source generator, add the test classes to the relevant [JsonSerializerContext declaration](https://github.com/dotnet/runtime/blob/3e960d90b1d1bc80f0c3890a237e526d01dca84b/src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/Serialization/PropertyVisibilityTests.cs#L594).

Thanks!

Reply:
And again amend: https://github.com/dotnet/runtime/commit/fd5475985189463f26b8643c0daa3f79d2595bcd